### ISSUE TITLE
False positive?

### ISSUE THREAD

**Issue by fabricesemti80** [ansible/ansible-lint#2386]
##### Summary

Logged an issue on Megalinter (https://github.com/oxsecurity/megalinter/issues/1828), but been directed to the linter by them.

I think this linter error is a false positive.

```sh
   INFO     Executing syntax check on provision/ansible/playbooks/cluster-prepare.yml (2.52s)
  WARNING  Listing 1 violation(s) that are fatal
  syntax-check: couldn't resolve module/action 'community.general.timezone'. This often indicates a misspelling, missing collection, or incorrect module path.
  provision/ansible/playbooks/cluster-prepare.yml:15:11 ERROR! couldn't resolve module/action 'community.general.timezone'. This often indicates a misspelling, missing collection, or incorrect module path.
  
  The error appears to be in '/github/workspace/provision/ansible/playbooks/cluster-prepare.yml': line 15, column 11, but may
  be elsewhere in the file depending on the exact syntax problem.
  
  The offending line appears to be:
  
        block:
          - name: Locale | Set timezone
            ^ here
  
  
  Error: syntax-check couldn't resolve module/action 'community.general.timezone'. This often indicates a misspelling, missing collection, or incorrect module path.
```

The config

```yaml
- hosts:
    - master
    - worker
  become: true
  gather_facts: true
  any_errors_fatal: true
  pre_tasks:
    - name: Pausing for 5 seconds...
      ansible.builtin.pause:
        seconds: 5
  tasks:
    - name: Locale
      block:
        - name: Locale | Set timezone
          community.general.timezone:
            name: "{{ timezone | default('Etc/UTC') }}"
```

 ( https://github.com/fabricesemti80/home-stack/blob/c74f88196f26213f2be095b548a884351011aab8/provision/ansible/playbooks/cluster-prepare.yml#L16) seems to be in line with the example (https://docs.ansible.com/ansible/latest/collections/community/general/timezone_module.html) on the module's site.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible-lint --version

```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### OS / ENVIRONMENT

Fedora 36 + k3s

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behavior

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:

- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
